superset/db_engine_specs/questdb.py

```diff
@@ -0,0 +1,311 @@
+        potential added grains but excluding any potentially disabled
+        grains in the config file.
+        :return: All time grain expressions supported by the engine
+        """
+        return cls._time_grain_expressions
+
+    @classmethod
+    def epoch_to_dttm(cls) -> str:
+        """SQL expression that converts epoch (seconds) to datetime that can be used in a
+        query. The reference column should be denoted as `{col}` in the return
+        expression, e.g. "FROM_UNIXTIME({col})"
+        :return: SQL Expression
+        """
+        return '{col} * 1000000'
+
+    @classmethod
+    def convert_dttm(cls, target_type: str, dttm: datetime, *_args, **_kwargs) -> Optional[str]:
+        """Convert a Python `datetime` object to a SQL expression.
+        :param target_type: The target type of expression
+        :param dttm: The datetime object
+        :return: The SQL expression
+        """
+        type_u = target_type.upper()
+        if type_u == 'DATE':
+            return f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
+        if type_u in ('DATETIME', 'TIMESTAMP'):
+            dttm_formatted = dttm.isoformat(sep=" ", timespec="microseconds")
+            return f"TO_TIMESTAMP('{dttm_formatted}', 'yyyy-MM-ddTHH:mm:ss.SSSUUUZ')"
+        return None
+
+    @classmethod
+    def get_datatype(cls, type_code: Any) -> Optional[str]:
+        """Change column type code from cursor description to string representation.
+        :param type_code: Type code from cursor description
+        :return: String representation of type code
+        """
+        return type_code.upper() if type_code and isinstance(type_code, str) else str(type_code)
+
+    @classmethod
+    def get_time_grains(cls) -> Tuple[TimeGrain, ...]:
+        """Generate a tuple of supported time grains.
+        :return: All time grains supported by the engine
+        """
+        return cls._engine_time_grains
+
+    @classmethod
+    def get_column_types(
+            cls,
+            column_type: Optional[str],
+    ) -> Optional[Tuple[TypeEngine, GenericDataType]]:
+        """Return a sqlalchemy native column type and generic data type that
+        corresponds to the column type defined in the data source (return None
+        to use default type inferred by SQLAlchemy). Override `column_type_mappings`
+        for specific needs (see MSSQL for example of NCHAR/NVARCHAR handling).
+        :param column_type: Column type returned by inspector
+        :return: SQLAlchemy and generic Superset column types
+        """
+        if not column_type:
+            return None
+        for regex, sqla_type, generic_type in cls._default_column_type_mappings:
+            matching_name = regex.search(column_type)
+            if matching_name:
+                return (
+                    types.resolve_type_from_name(sqla_type.__visit_name__).impl,
+                    generic_type
+                )
+        return None
```

Comment:
and here

Reply:
correct! 9a3c588dc495ad0435666f41b455d54ee57b40fc